### PR TITLE
CI: Add support in yml for detecting darc dependency changes in `eng/Version.Details.xml`

### PR DESCRIPTION
**Pull request by radical**
## Issue

Currently, whenever a darc flow PR is opened, we cannot trigger different jobs based on which
dependency got updated. This is needed because in some cases, like for ILLinker, Emscripten
updates we want to trigger all the wasm jobs. But that might not be needed for some of the other
dependencies.

- This causes failures to slip through to `main`, which gets discovered after the fact as other PR
failures, or rolling build failures, creating extra work for developers.

## Solution

- This PR identifies the changed dependencies, and emits one azure variable per dependency,
which can then be used in conditions in the yml .
- The changed dependency can be checked as `dependencies.evaluate_paths.outputs['DarcDependenciesChanged.System_CommandLine']`

- Included this in the `evaluate_paths` job itself, instead of creating another job

- Also, use this for wasm jobs

### CODE REVIEW

**Comment by radical** [comment]
@akoeplinger thoughts? The changes right now include an artificial change to `eng/Version.Details.xml`.

1. In `Evaluate Paths` job, two variables get set, one per changed dependency:

```
eng/pipelines/evaluate-changed-darc-deps.sh --difftarget HEAD^1
========================== Starting Command Output ===========================
/usr/bin/bash --noprofile --norc /home/vsts/work/_temp/45925021-bbde-4179-bee2-f8f8b3eda8da.sh
Setting pipeline variable Microsoft_NETCore_Runtime_ICU_Transport=true
Setting pipeline variable System_CommandLine=true
Finishing: Evaluate eng/Version.Details.xml for dependency changes
```

2. And I tried accessing the variable in WBT job:
```
  CommandLineChanged1:
    Parsing expression: <dependencies.evaluate_paths.outputs['DarcDependenciesChanged.System_CommandLine']>
    Evaluating: dependencies['evaluate_paths']['outputs']['DarcDependenciesChanged.System_CommandLine']
    Result: 'true'
```

File: eng/pipelines/common/variables.yml [hunk 1]
```diff
@@ -34,4 +34,11 @@ variables:
   ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
     value: Debug
 
+- name: wasmDarcDepsChanged
+  value: $[ or(
+              eq(dependencies.evaluate_paths.outputs['DarcDependenciesChanged.Microsoft_NET_Workload_Emscripten_Manifest-7_0_100'], true),
```
Comment (akoeplinger) [inline eng/pipelines/common/variables.yml:39]: can we somehow do a string contains check here so we don't need to include the `7_0_100` version number? we'll almost certainly forget to update this here.

File: eng/pipelines/common/variables.yml [hunk 2]
```diff
@@ -34,4 +34,11 @@ variables:
   ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
     value: Debug
 
+- name: wasmDarcDepsChanged
```
Comment (akoeplinger) [inline eng/pipelines/common/variables.yml:37]: nit: I'd prefer a more explicit `wasmDarcDependenciesChanged` name

**Review by akoeplinger — APPROVED** [review]
LGTM but I'd prefer to get someone from the infrastructure team to take a look as well

File: eng/pipelines/common/variables.yml (hunk 1 above)
Comment (radical) [reply eng/pipelines/common/variables.yml:39]: yeah, I'll try

**Comment by radical** [comment]
`Build windows arm64 Release NativeAOT` is https://github.com/dotnet/runtime/issues/69832 .
`Libraries Build windows x64 Debug` is https://github.com/dotnet/runtime/issues/73646 .
`Libraries Test Run checked coreclr Linux arm64 Debug` is https://github.com/dotnet/runtime/issues/73647
`Libraries Test Run release coreclr OSX x64 Debug` failed due to network issue in downloading tests
`mono llvmfullaot Pri0 Runtime Tests Run Linux x64 release` is  https://github.com/dotnet/runtime/issues/69832 .

And all of these are unrelated to this PR.

**Pull request merged**